satellite/src/main/java/com/linbit/linstor/layer/dmsetup/DmSetupUtils.java

```diff
@@ -187,6 +187,76 @@ public static Set<String> list(ExtCmd extCmd, String target) throws StorageExcep
         return ret;
     }
 
+    public static void fixSymlinkForDevice(ExtCmd extCmd, String deviceName) throws StorageException
+    {
+        try
+        {
+            OutputData outputData = extCmd.exec("dmsetup", "ls");
+            ExtCmdUtils.checkExitCode(
+                outputData,
+                StorageException::new,
+                "listing devices from dmsetup ls failed "
+            );
+
+            String stdOut = new String(outputData.stdoutData);
+            Matcher matcher = DM_SETUP_LS_PATTERN.matcher(stdOut);
+            String dmName = resolveDeviceName(deviceName);
+            String minor = null;
+
+            while (matcher.find())
+            {
+                String devName = matcher.group(1);
+                minor = matcher.group(3);
+
+                if (devName.equals(dmName))
+                {
+                    break;
+                }
+            }
+
+            if (minor == null)
+            {
+                throw new StorageException(
+                    "Device \"" + dmName + "\" (generated from \"" + deviceName + "\") not found in dmsetup output:\n\n" + stdOut
+                );
+            }
+
+            // Create symlink
+            OutputData symlinkOutput = extCmd.exec("ln", "-s", "../dm-" + minor, deviceName);
```

Comment:
Honestly I am not a big fan of "manually" creating symlinks in `/dev/${lvm_volume_group}/...`. That sounds like it should be done via a UDEV-rule or LVM itself, but certainly not by Linstor.

Reply:
It should be done by udev, but there are some certain issues which might affect that symlinks are disappearing.
We have many cases on various systems, especially old ubuntus and etc. These cases confusing linstor and make it stuck in the middle of some operation, eg. resizing.

As an alternative there is `vgscan --mknodes` but it has disadvantages over creating symlinks manually:
- It is deprecated and not recomended to use
- It creates character devices but not symlinks
- You can not pass arguments to handle specific devices only

Reply:
I know that it is not LINSTOR problem. For the same reason I set WARN level for logging such issues. Thus now LINSTOR will not fail in these cases, but user will see warning messages.

Reply:
I agree with @kvaps, we regularly get issues with missing symlinks. I understand that this is not a Linstor problem, however, we urgently needs this hack.